### PR TITLE
[redesign][governance] proposal details view.

### LINKED ISSUE #3288: Governance Proposals Cards update
You can find all updated layouts, components sheet, colors palette, basically everything in Figma https://www.figma.com/file/EanUbqJmPs8EMqY26j9zwo/decred-piui-master?node-id=12%3A0

There are 2 types of a proposal description card: when we can vote or we can’t.

When we can vote: _**Voting tab**_.

The structure: 3 sections
1st -  a card design repeats, but the arrow that has been on the right were moved to the left and it means “Go back to the list of proposals”.
2nd - voting section. When you click on a preferable radio button, a sign “Voted” and a checkmark appears. Previously you had to vote manually by clicking on a Vote button, but now your vote can be accepted right after clicking on the right radio button. After clicking on it "Voted" text and a checkmark displays for a few seconds. Same for Governance Changes cards.
3rd - proposal description.
And “See Proposal Comments on Politeia” button.

**How it's implemented:(it's "finished voting", but the concept is the same)**
image

**New designs:**
image

**Voted**
image

You can click on the name of a proposal and be redirected to it on the Politeia website.
image

When we can’t vote: **_In Discussion, Finished voting, Abandoned_** tabs.

The structure is the same as for Voting tab, but there is no voting section.
To go back to the list of proposals you have to click on the arrow within the 1st section.

**Implemented:**
image

**New designs:**
image

### PR DESCRIPTION
**Pull request by amassarwi**
Main changes:

- Split `ProposalsDetials.module.css` helpers styling into components own css modules.
-  Add vote end event & blocks left number.
- Define proposal status & vote status as constants.
- Add Join component to display subtitle separated fields.
- Add global helper css classes.
- Ditch `OverviewField` component.
- Replace `EligibleTickets` component with new `VoteSection`.
- Delete `TxHsitory/EligibleRow.jsx` and it's styling.
- Move Vote Pref. component to `VoteSection` according to new specs.
- Improve viewedProposal state obj to avoid duplicated fields.
- Declarative Proposals Details Page helpers.
- Replace shared classes with new components (ie `VoteButton` & `NoTicketsMsg`)
-  Finish Vote Preference section & introduce `VotedCheckmark`.
- Rename ChooseOptions component to VotePreference.
- Rename ChooseVoteOption to VotePreferenceWrapper.
- Move CastVoteModalButton to a separate dir with css module.
- Tweak proposal body & politeia proposal button/link.
- Re-arrange proposal card.
- Tweak subtitle.
- Add proposal card back button.
- Add title tooltip.

**Note:** before reviewing the code I'd suggest comparing `components/viewed/ProposalDetailsPage` dir and it's structure on master and on this branch.

Screenshots:

Voting proposal & user voted:
image

Abandoned:
Screenshot 2021-04-05 at 3 11 40

Rejected:
Screenshot 2021-04-05 at 3 11 53

Closes #3288.

### CODE REVIEW

**Comment by amassarwi** [comment]
Tested and found two issues:

1. Approved proposals showing currently rejected status tag.
2. Minor alignment issue in vote section:
image

**Comment by amassarwi** [comment]
**Note:** This was implemented & tested against pre-tlog poltieia code, I'll address the two issues listed above 
and then will test it against latest politeia commit and see how it plays.

File: app/components/views/ProposalDetailsPage/ProposalDetailsPage.jsx [hunk 1]
```diff
@@ -32,18 +32,19 @@ const ProposalDetailsPage = () => {
           </div>
         );
       case "success":
+        // XXX this should move to redux logic, we should parse proposal body
```
Comment (amassarwi) [inline app/components/views/ProposalDetailsPage/ProposalDetailsPage.jsx:35]: will followup with a PR for this `XXX`

File: app/components/views/ProposalDetailsPage/helpers/VotePreferenceWrapper/VotePreferenceWrapper.jsx [hunk 2]
```diff
@@ -17,13 +17,14 @@ const getError = (error) => {
...
-const ChooseVoteOption = ({
+const VotePreferenceWrapper = ({
   viewedProposalDetails,
   voteOptions,
   currentVoteChoice,
-  votingComplete,
-  eligibleTicketCount
+  votingComplete
 }) => {
+  // XXX move all hooks/state logic to a custom hook ie. useVotePreference
```
Comment (amassarwi) [inline app/components/views/ProposalDetailsPage/helpers/VotePreferenceWrapper/VotePreferenceWrapper.jsx:26]: will do in this PR

File: app/constants/index.js [hunk 3]
```diff
@@ -1,5 +1,7 @@
+// XXX change all files to small letters!
```
Comment (amassarwi) [inline app/constants/index.js:1]: will followup with this change.

**Comment by amassarwi** [comment]
> Tested and found two issues:
> 
>     1. Approved proposals showing currently rejected status tag.
> 
>     2. Minor alignment issue in vote section:
>        image

1. Looks like BE returns in votesummary `approved: false` for the proposal in the screenshot, others have the right
approved tag with latest commit:
Screenshot 2021-04-05 at 4 10 40

2. Improved.

**Comment by amassarwi** [comment]
Need few last touches in smaller screens.

**Comment by alexlyp** [comment]
Needs a rebase

**Comment by amassarwi** [comment]
@alexlyp rebased & improved smaller screens as well.

**Comment by alexlyp** [comment]
Getting an error when I go to the top result in the Finished category (Underwater Citadel):
hooks.js:16 Uncaught TypeError: Cannot destructure property 'viewedProposalDetails' of 'undefined' as it is undefined.

This wallet has no live tickets currently if that helps reproduce.

**Comment by amassarwi** [comment]
@alexlyp updated

**Comment by alexlyp** [comment]
Seeing some text in dark theme show as white on white.  Voting power and the short hash below the status on the proposal details page

**Review by alexlyp — APPROVED** [review]

**Pull request merged**